libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -1539,6 +1565,35 @@ bool AP_AHRS::set_origin(const Location &loc)
     return success;
 }
 
+void AP_AHRS::record_origin()
+{
+    _origin_lat.set_and_save_ifchanged(state.origin.lat * 1.0e-7);
+    _origin_lng.set_and_save_ifchanged(state.origin.lng * 1.0e-7);
+    _origin_alt.set_and_save_ifchanged(state.origin.alt * 1.0e-2);
+}
+
+bool AP_AHRS::ensure_origin_is_set()
+{
+    if (state.origin_ok) {  // we already have an origin
+        return true;
+    }
+
+    const Location loc = {
+        int32_t(_origin_lat.get() * 1e7),
+        int32_t(_origin_lng.get() * 1e7),
+        int32_t(_origin_alt.get() * 100),
+        Location::AltFrame::ABSOLUTE
+    };
+
+    if (set_origin(loc)) {
```

Comment:
OK, so this structure means we can set the origin to 0,0,0,0

We're supposed to be able to cope with that, but if you do you end up with:

```
LOITER> 171: LOCAL_POSITION_NED {time_boot_ms : 433864, x : -3936597.5, y : 15820468.0, z : -670.9725341796875, vx : 0.009984244592487812, vy : -0.02085096575319767, vz : -2.4886622428894043}
```

If you want fine local position this is probably not going to end well, depending on what estimator you're using, anyway.

Reply:
I'm inclined to make the defaults the GPS co-ordinates of Spring Valley - what do you think?